vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthCombinedLoginFragment.kt

```diff
@@ -70,6 +74,28 @@ class FtueAuthCombinedLoginFragment :
             viewModel.handle(OnboardingAction.UserNameEnteredAction.Login(views.loginInput.content()))
         }
         views.loginForgotPassword.debouncedClicks { viewModel.handle(OnboardingAction.PostViewEvent(OnboardingViewEvents.OnForgetPasswordClicked)) }
+
+        viewModel.onEach(OnboardingViewState::canLoginWithQrCode) {
```

Comment:
Maybe to be moved inside invalidate method?

Reply:
Not possible probably since this is an abstract class.

Reply:
It is because the `invalidate` method has been declared `final` in the `AbstractFtueAuthFragment`. I wonder why it has been declared `final`.